File: app/src/main/java/com/temperoni/recipes/domain/event/ResponseEvent.java
package com.temperoni.recipes.domain.event;

/**
 * @author Leandro Temperoni
 */

public class ResponseEvent<T> {

    private boolean success;
    private T payload;
    private String errorMessage;

    public boolean isSuccess() {
        return success;
    }

    public T getPayload() {
        return payload;
    }

    public void setPayload(T payload) {
        this.payload = payload;
        success = true;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
        success = false;
    }
}

File: app/src/main/java/com/temperoni/recipes/domain/event/ResponseEvent.kt
package com.temperoni.recipes.domain.event

/**
 * @author Leandro Temperoni
 */
open class ResponseEvent<T> {

    var isSuccess: Boolean = false

    var payload: T? = null
        set(payload) {
            field = payload
            isSuccess = true
        }

    var errorMessage: String? = null
        set(errorMessage) {
            field = errorMessage
            isSuccess = false
        }
}
